[[ Bug 19059 ]] Add access to LCS delimiter local properties to LCB

runrevmark opened the pull request:
This patch adds the following syntax to LCB when running in the
LC engine:

   the line delimiter
   the item delimiter
   the row delimiter
   the column delimiter

These give access to the value of the delimiter properties in the
most recent LCS handler on the stack; or sensible defaults if there
is no LCS handler on the stack.

runrevmark committed:
[[ Bug 19059 ]] Add access to LCS delimiter local properties to LCB

This patch adds the following syntax to LCB when running in the
LC engine:

   the line delimiter
   the item delimiter
   the row delimiter
   the column delimiter

These give access to the value of the delimiter properties in the
most recent LCS handler on the stack; or sensible defaults if there
is no LCS handler on the stack.

peter-b commented on engine/src/engine.lcb:
```diff
@@ -406,4 +411,68 @@ begin
+
+Description:
+Fetches the calling (script) handler's columnDelimiter property. If there is no
+script context, then it returns tab.
```
Some general points for all the documentation blocks! Please:

* use inline code markup where applicable (e.g. for `columnDelimiter`) in this example
* make sure examples can be compiled and run if they're simply copy-and-pasted into a source file with no further alterations, and show what the result is likely to be. E.g.

        Examples:
            -- Split a string into a list using the column delimiter
            variable tItems
            put "alice\tbob\teve" into tItems
            split tItems by the column delimiter
            -- tItems is now a list: ["alice", "bob", "eve"]

* add cross-references to related documentation items, and appropriate tags

Also, in this description, "it" is unclear.  Consider something like:

> Fetch the calling (script) handler's `columnDelimiter` property, or horizontal tab (U+0009) if no script context is available.

peter-b commented on tests/lcs/core/engine/extension.livecodescript:
```diff
@@ -161,3 +161,19 @@ on TestExtensionLibraryHandlerCallErrors
+
+   TestDiagnostic tDelimiters
+   TestAssert "library handler can access delimiters", tDelimiters is "a,b,c,d"
+end TestExtensionLibraryDelimiterAccess
```
Is there any way to add a test covering the "no script context available" case?

runrevmark replied:
Not at present no - there's no easy way to get LCB code to execute 'out of context' - it would need to be a handler which runs directly from the invocation of the root event loop.

peter-b reviewed (changes requested):
Please add appropriate tags and cross-references to the documentation comments for the syntax!

runrevmark committed:
[[ Bug 19059 ]] Improve documentation entries

This patch adds proper examples to each of the delimiter properties,
and makes sure they are all related to each other.

runrevmark committed:
[[ Bug 19059 ]] Add tags to all syntax in engine.lcb

This patch adds the 'Script Engine' tag to all the syntax defined
in the engine LCB module.

runrevmark commented:
@peter-b: Can haz review? :)

peter-b commented:
@livecode-vulcan review ok 9c8bb82

livecode-vulcan commented:
💙 review by @peter-b ok 9c8bb824f1e0643896ca01119eda14e5c76d511d

livecode-vulcan commented:
😎 test success 9c8bb824f1e0643896ca01119eda14e5c76d511d

* try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/889)
* try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/895)
* try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/887)
* try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/888)
* try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/895)
* try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/886)
* try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/886)
